sklearn/metrics/tests/test_score_objects.py

```diff
@@ -480,39 +484,45 @@ def test_scorer_sample_weight():
     # Unlike the metrics invariance test, in the scorer case it's harder
     # to ensure that, on the classifier output, weighted and unweighted
     # scores really should be unequal.
-    X, y = make_classification(random_state=0)
+    X, y = make_classification(n_samples=101, random_state=0)
     _, y_ml = make_multilabel_classification(n_samples=X.shape[0],
                                              random_state=0)
+    _, y_reg = make_regression(n_samples=X.shape[0], n_features=X.shape[1],
```

Comment:
It seems that `test_scorer_sample_weight` was designed to test classification score.

I think it would be best to create 2 tests function one dedicated to classification and one dedicated to regression. It will be easier to follow what is going on.